File: Retrofit/app/src/main/java/com/example/arnassmicius/retrofit/GitHubRepoAdapter.java
package com.example.arnassmicius.retrofit;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.arnassmicius.retrofit.api.model.GitHubRepo;

import java.util.List;

/**
 * Created by arnas on 18.2.16.
 */

public class GitHubRepoAdapter extends RecyclerView.Adapter<ReposAdapterViewHolder> {

    private Context context;
    private List<GitHubRepo> repos;

    public GitHubRepoAdapter(Context context, List<GitHubRepo> repos) {
        this.context = context;
        this.repos = repos;
    }

    @Override
    public ReposAdapterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.list_item_repo, parent, false);
        return new ReposAdapterViewHolder(view);
    }

    @Override
    public void onBindViewHolder(ReposAdapterViewHolder holder, int position) {
        holder.listItem.setText(repos.get(position).getName());
    }

    @Override
    public int getItemCount() {
        return repos.size();
    }
}

File: Retrofit/app/src/main/java/com/example/arnassmicius/retrofit/GitHubRepoAdapter.kt
package com.example.arnassmicius.retrofit

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.example.arnassmicius.retrofit.api.model.GitHubRepo

/**
 * Created by arnas on 18.2.16.
 */

class GitHubRepoAdapter(private val context: Context, private val repos: List<GitHubRepo>) : RecyclerView.Adapter<ReposAdapterViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReposAdapterViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.list_item_repo, parent, false)
        return ReposAdapterViewHolder(view)
    }

    override fun onBindViewHolder(holder: ReposAdapterViewHolder, position: Int) {
        holder.listItem.text = repos[position].name
    }

    override fun getItemCount(): Int {
        return repos.size
    }
}
